### DISCUSSION TITLE
Suggestion! exclude NSFW words from auto generated typing text.

### DISCUSSION THREAD

**Discussion by okpreetam** [monkeytypegame/monkeytype#1581] [Ideas] [answered]
Recently I'm doing monkeytype and tackle the word "**UPSKIRT**". As English is my second language I know that this word is not appropriate for this platform, or maybe I do not know the correct meaning of this word so I googled it and got some awkward results 😂. This is the first time I face such a word in my 28 hours of typing time.

**So It is good to blacklist some NSFW words, at least for people below 18.**

> It would be good if we comment down such words. so it easy for contributors to rectify them.

**Comment by unrealapex** [comment, marked as answer]
What language where you doing(`english`, `english 1k`, `english 10k`, etc...) when you encountered this word?  There were and still might be more words that would be considered not appropriate in the language json files. I assume the users who contributed these files added the words in good faith(probably copy pasting from a website like Wikipedia, Project Gutenberg, and others) and failed to catch the word(s) before submitting the PR, or just didn't review it at all. The contributors of sites that contain these words probably added these words with malicious intent. Even well known products like the [Merriam-Webster dictionary](https://www.merriam-webster.com) have offensive words in them. Coincidentally, while I was writing this answer, I wanted to find a synonym for the word bad, looking at an online dictionary(Merriam-Webster) I found an [offensive word](https://www.merriam-webster.com/thesaurus/bad)(one of the synonyms for bad) I am very sure "Niggardly" shouldn't be in the dictionary or anywhere else. I don't know if I can report it in any way. I can remove it.

**Reply by okpreetam** [reply]
Only present in english_10k.json, I searched all English language JSON files

![image](https://user-images.githubusercontent.com/38487176/123992918-a3952500-d9e9-11eb-9fbc-e34f61b01ed7.png)
.

**Reply by unrealapex** [reply]
#1584
